service/controller/clusterapi/v29/cluster_resource_set.go

```diff
@@ -243,8 +177,9 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 	var encryptionResource controller.Resource
 	{
 		c := encryption.Config{
-			Encrypter: encrypterObject,
-			Logger:    config.Logger,
+			Encrypter:     encrypterObject,
+			Logger:        config.Logger,
+			ToClusterFunc: key.ToCluster,
```

Comment:
The encryption resource works with the `Cluster` type. Reworking it to make it abstract and work with also `MachineDeployment` types was not a good option for me now. So I went with the same approach we have in `cluster-operator` where we use a lookup function. In the cluster controller it is simply the key function. In the machine deployment controller it is a real lookup function using the CMA client.